syntax="proto3";

package client_handler;
service ClientHandler {
	rpc GetStatus(GetRequest) returns (stream StatusResponse){}
	rpc GetConnectConfig(GetRequest) returns (stream ConfigResponse){}
	rpc GetConnectQR(GetRequest) returns (stream ConfigResponse){}
	rpc HandleConnect(GetRequest) returns (stream ConnectResponse){}
}


message GetRequest {
	string access_token = 1;
}

message StatusResponse {
	oneof response {
		Ack ack = 1;
		InfoResponse info = 2;
	}
}

message Ack {
	string message = 1;
}

message InfoResponse {
	bool status = 1;
}

message ConfigResponse {
	oneof response {
		Ack ack = 1;
		InfoResponse info = 2;
	}
}

message Image {
	string image_data=1;
}

message ConnectResponseInfo {
	bool status = 1;
}

message ConnectResponse {
	oneof response {
		Ack ack = 1;
		ConnectResponseInfo info = 2;
	}
}
